src/ophyd_async/epics/_backend/common.py

```diff
@@ -1,25 +1,25 @@
 from enum import Enum
-from typing import Any, Optional, Tuple, Type
+from typing import Dict, Optional, Tuple, Type
 
 
-def get_supported_enum_class(
+def get_supported_values(
     pv: str,
-    datatype: Optional[Type[Enum]],
-    pv_choices: Tuple[Any, ...],
-) -> Type[Enum]:
+    datatype: Optional[Type[str]],
+    pv_choices: Tuple[str, ...],
+) -> Dict[str, str]:
     if not datatype:
-        return Enum("GeneratedChoices", {x or "_": x for x in pv_choices}, type=str)  # type: ignore
+        return {x: x or "_" for x in pv_choices}
 
-    if not issubclass(datatype, Enum):
-        raise TypeError(f"{pv} has type Enum not {datatype.__name__}")
     if not issubclass(datatype, str):
-        raise TypeError(f"{pv} has type Enum but doesn't inherit from String")
-    choices = tuple(v.value for v in datatype)
-    if set(choices) != set(pv_choices):
-        raise TypeError(
-            (
-                f"{pv} has choices {pv_choices}, "
-                f"which do not match {datatype}, which has {choices}"
+        raise TypeError(f"{pv} is type Enum but doesn't inherit from String")
+    if issubclass(datatype, Enum):
+        choices = tuple(v.value for v in datatype)
+        if choices != pv_choices:
+            raise TypeError(
+                (
+                    f"{pv} has choices {pv_choices}, "
+                    f"which do not match {datatype}, which has {choices}"
+                )
             )
-        )
-    return datatype
+        return {x: datatype(x) for x in pv_choices}
+    return {x: x for x in pv_choices}
```

Comment:
I literally have no idea how this is working, here you drop the reference to the Enum, and keep only its string value...

Reply:
```
if issubclass(datatype, Enum):
        choices = tuple(v.value for v in datatype)
        if choices != pv_choices:
            raise TypeError(
                (
                    f"{pv} has choices {pv_choices}, "
                    f"which do not match {datatype}, which has {choices}"
                )
            )
        return {x: datatype(x) for x in pv_choices}  # <- This is if the datatype is enum, we make a map of str : Enum
    return {x: x for x in pv_choices}  # <- This is if the datatype is not None, not an Enum, is str or some other str extending type. We make a map of str: str
```